pennylane/ops/functions/equal.py

```diff
@@ -188,3 +228,12 @@ def _equal_shadow_measurements(
     k_match = op1.k == op2.k
 
     return return_types_match and wires_match and H_match and k_match
+
+
+def _obs_comparison_data(obs):
```

Comment:
From the `Possible Drawbacks:` section:
> This makes comparison of two Hermitian operators slightly worse, since they are now compared as Observables, which checks for an exact match instead of using math.allclose to compare the elements of the matrix.

This sounds like a breaking change, is it something that aligns with our plans? Preserving the existing behaviour could be done by creating a new case for `qml.Hermitian`, no? Given that we'd like to do that.

Also, a keyword argument could be added potentially if we wanted more options.

Reply:
I think defaulting to the 'old' behaviour for Hermitian makes sense. I can't see any reason for it to be based on a kwarg, it should just continue to compare Hermitians as it has. I can add that.